src/openforms/submissions/api/serializers.py

```diff
@@ -265,18 +264,18 @@ def validate_data(self, data: dict):
         return data
 
     def _run_formio_validation(self, data: dict) -> None:
+        from ..form_logic import evaluate_form_logic
+
         # Check feature flag to opt out of formio validation first.
         config = GlobalConfiguration.get_solo()
         assert isinstance(config, GlobalConfiguration)
         if not config.enable_backend_formio_validation:
             return
 
-        # take a deep copy because we will be mutating each component validate.required
-        # to massage the validation serializer into a more "relaxed" variant.
-        assert self.instance.form_step is not None
-        fd = self.instance.form_step.form_definition
         submission = self.instance.submission
-        configuration = copy.deepcopy(fd.configuration)
+        # evaluate dynamic configuration
+        configuration = evaluate_form_logic(submission, step=self.instance, data=data)
+
         # mark them all as not required
```

Comment:
I currently remember why this was done in https://github.com/open-formulieren/open-forms/pull/4144, but I might forgot in the future. Maybe this comment can be updated to something like:
```suggestion
        # mark them all as not required, to support pausing forms (where data can still be imcomplete)
```